From 4d92469a71290f69fb2848f9455276aff8e36905 Mon Sep 17 00:00:00 2001
From: Mandy Hung <mandy.hung.wiwynn@gmail.com>
Date: Fri, 21 Nov 2025 08:24:37 +0000
Subject: [PATCH 16/16] oem-meta: Add missing propertyName for sled cycle

Add the following in the sled cycle section:
dbusMapping.propertyName = "RequestedPowerTransition"

Motivation:
Without the propertyName, the D-Bus call for the sled cycle command
hits an UnknownProperty error, preventing the sled cycle command from
executing successfully.

Change-Id: I8293d08523b76cc61bd6d139abc6bd12142a5389
Signed-off-by: Mandy Hung <mandy.hung.wiwynn@gmail.com>
Signed-off-by: Eric Yang <eric.yang.wiwynn@gmail.com>
---
 oem/meta/libpldmresponder/file_io_type_power_control.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/oem/meta/libpldmresponder/file_io_type_power_control.cpp b/oem/meta/libpldmresponder/file_io_type_power_control.cpp
index 32de91d..da8b28b 100644
--- a/oem/meta/libpldmresponder/file_io_type_power_control.cpp
+++ b/oem/meta/libpldmresponder/file_io_type_power_control.cpp
@@ -43,6 +43,7 @@ int PowerControlHandler::write(const message& data)
             dbusMapping.objectPath =
                 std::string("/xyz/openbmc_project/state/chassis0");
             dbusMapping.interface = "xyz.openbmc_project.State.Chassis";
+            dbusMapping.propertyName = "RequestedPowerTransition";
             property =
                 "xyz.openbmc_project.State.Chassis.Transition.PowerCycle";
             break;
-- 
2.47.3

